hermes/modules/vectorgenerator.py

```diff
@@ -0,0 +1,147 @@
+# MovieLens
+# ================================================================================
+
+class MovieLens(object):
+	@classmethod
+	def isSameDataInstance(cls, comparisonData):
+		return comparisonData.vectorizer == "movielens"
+
+class MovieLensUserVector(UserVector, MovieLens):
+	def ratings(self):
+		return self.data.dataframe.map(lambda row: (row.user_id, row.movie_id, row.rating))
+
+	def pos_ratings(self):
+		return self.data.dataframe.map(lambda row: (row.user_id, row.movie_id, row.rating)).filter(lambda (u, m, r): r > 3)
+
+	def ratings_to_interact(self):
+		return self.data.dataframe.map(lambda row: (row.user_id, row.movie_id, -1 if row.rating < 3 else 1))
+
+class MovieLensContentVector(ContentVector, MovieLens):
+	def genre(self):
+		def genre_vectorizer(row):
+			return np.array((
+					int(row.genre_action),
+					int(row.genre_adventure),
+					int(row.genre_animation),
+					int(row.genre_childrens),
+					int(row.genre_comedy),
+					int(row.genre_crime),
+					int(row.genre_documentary),
+					int(row.genre_drama),
+					int(row.genre_fantasy),
+					int(row.genre_filmnoir),
+					int(row.genre_horror),
+					int(row.genre_musical),
+					int(row.genre_mystery),
+					int(row.genre_romance),
+					int(row.genre_scifi),
+					int(row.genre_thriller),
+					int(row.genre_war),
+					int(row.genre_western),
+				))
+		return self.data.dataframe.map(lambda row: (row.movie_id, ))
+
+# ================================================================================
+# Wiki
+# ================================================================================
+
+class Wiki(object):
+	@classmethod
+	def isSameDataInstance(cls, comparisonData):
+		return comparisonData.vectorizer == "wiki"
+
+class WikiUserVector(UserVector, Wiki):
+	def __init__(self):
+		super(self.__class__, self).__init__()
+		self.filtered = self.sqlCtx.sql("select * from ratings where redirect_target is null and article_namespace=0 and user_id is not null")
+		self.filtered.registerTempTable("wiki_ratings")
+
+	def num_edits(self):
+		return self.sqlCtx.sql("select user_id as user, article_id as item, count(1) as rating from wiki_ratings group by user_id, article_id")
+
+	def any_interact(self):
+		return self.sqlCtx.sql("select user_id as user, article_id as item, 1 as rating from wiki_ratings group by user_id, article_id")
+
+	def num_edits_ceil(self):
+		return self.sqlCtx.sql("select user_id as user, article_id as item, count(1) as rating from wiki group by user_id, article_id")\
+		     .map(lambda (user, article, rating): (user, article, max(rating, 5)))
+
+class WikiContentVector(ContentVector, Wiki):
+	def __init__(self):
+		super(self.__class__, self).__init__()
+		self.filtered_content = sqlCtx.sqlCtx.sql("select * from content where redirect_target is null and article_namespace=0 and full_text is not null")
+		self.filtered_content.registerTempTable("wiki_content")
+
+	def glove(self):
+		pass
```

Comment:
NotImplementedError is the right way to go. I put pass temporarily since glove is already implemented, and I just wanted to leave a placeholder function there for future implementation.